crates/sncast/tests/helpers/fixtures.rs

```diff
@@ -67,6 +76,13 @@ pub async fn declare_contract(account: &str, path: &str, shortname: &str) -> Fie
     );
 
     let tx = declaration.send().await.unwrap();
+
+    // scarb clean does not remove lockfile, and we want to do that
+    fs::remove_file(format!("{}/{}", contract_dir, "Scarb.lock"))
+        .expect("Failed to remove test contract's lockfile");
+    fs::remove_dir_all(format!("{}/{}", contract_dir, "target"))
+        .expect("Failed to remove test contract's target dir");
```

Comment:
Instead of removing artifacts here, it would be better to copy the source file of the contract to a temporary directory and build it there. Then, the files created would be automatically deleted once the temp dir goes out of scope.